### PR TITLE
DESMAKERS-3827: enable 11/29Bits id frame send and receive

### PR DESCRIPTION
**Pull request by LinjingZhang**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Now sendpacket() support both 11 & 29 bits identifier. same for receive. 
Functions that are not the same as the official API can be removed, updating the README and example sketches.

Related Issue
N/A

Context
Once it's up and running, it works ;)

### CODE REVIEW

**Review by jaenrig-ifx — APPROVED** [review]
As usual, minor clue about the CAN details. But if the corresponding (manual) tests are performed. 
I can just approve :)

File: libraries/CAN/src/CANXMC.cpp [hunk 1]
```diff
@@ -48,12 +48,14 @@ CANXMC::~CANXMC() {}
         .sample_point = (uint16_t)(80 * 100),
         .sjw = (uint16_t)1,
     };
+
     XMC_CAN_Enable(CAN_xmc);
+
      /* Configuration of CAN Node and enable the clock */ 
     #if (UC_FAMILY == XMC4)
-      #define CAN_CLOCK_SOURCE ((XMC_CAN_CANCLKSRC_t)XMC_CAN_CANCLKSRC_FPERI) 
+      #define CAN_CLOCK_SOURCE ((XMC_CAN_CANCLKSRC_t)XMC_CAN_CANCLKSRC_FPERI) // synchronous clock for XMC4000
```
[the remark this answers is not in the document]
Comment (LinjingZhang) [reply libraries/CAN/src/CANXMC.cpp:56]: Yes, I agree - the XMC4000 and XMC1000 have some different settings. But also not board different... Maybe in the future a more reasonable set of variants could be considered

[the remark this answers is not in the document]
Comment (OlafFilies) [reply libraries/CAN/src/CANXMC.cpp:56]: maybe this should be defined in variants files

**Review by OlafFilies — APPROVED** [review]
look good for me, but please test on hardware.

File: libraries/CAN/src/CANXMC.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (LinjingZhang) [reply libraries/CAN/src/CANXMC.cpp]: I changed the variants files 😄

**Pull request merged**